server/config.go

```diff
@@ -0,0 +1,59 @@
+package server
+
+import (
+	"github.com/ns1labs/orb/datasource"
+)
+
+type ErrorLogLevel int
+
+const (
+	Error ErrorLogLevel = iota
+	Warn
+	Info
+)
+
+type DataStore struct {
+	StoreType string `mapstructure:"type"`
+}
+
+type Services struct {
+	DataStore datasource.ConsulConfig `mapstructure:"datasource"`
+}
+
+type AutoTLS struct {
+	Enabled      bool   `mapstructure:"enabled"`
+	Domain       string `mapstructure:"domain"`
+	CertCacheDir string `mapstructure:"cache_dir"`
+}
+
+type TLSConfig struct {
+	AutoTLS AutoTLS `mapstructure:"auto"`
+	Cert    string  `mapstructure:"cert"`
+	Key     string  `mapstructure:"key"`
+}
+
+type Listener struct {
+	BindAddr  string    `mapstructure:"bind_addr"`
+	TLSConfig TLSConfig `mapstructure:"tls"`
+}
+
+type ErrorLog struct {
+	Level string `mapstructure:"level"`
+	Path  string `mapstructure:"path"`
+}
+
+type AccessLog struct {
+	Format string `mapstructure:"format"`
+	Path   string `mapstructure:"path"`
+}
+
+type Loggers struct {
+	ErrorLog  ErrorLog  `mapstructure:"error_log"`
+	AccessLog AccessLog `mapstructure:"access_log"`
+}
+
+type Config struct {
```

Comment:
I added this config to give the config file structure that is enforced. So we aren't really using viper to its fullest here because it can query the configs at runtime if we want. Im not sure which way we prefer to go here. Either way my plan was not to pass this config down into the api handlers and just let the server router assemble the needed components later.